python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -1013,6 +1023,79 @@ def cancel_job(self, job_id: int) -> Mapping[str, Any]:
             method="DELETE", endpoint=f"jobs/{job_id}", base_url=self.rest_api_base_url
         )
 
+    def sync_and_poll(
+        self,
+        connection_id: str,
+        poll_interval: Optional[float] = None,
+        poll_timeout: Optional[float] = None,
+        cancel_on_termination: bool = True,
+    ) -> AirbyteOutput:
+        """Initializes a sync operation for the given connection, and polls until it completes.
+
+        Args:
+            connection_id (str): The Airbyte Connection ID. You can retrieve this value from the
+                "Connection" tab of a given connection in the Airbyte UI.
+            poll_interval (float): The time (in seconds) that will be waited between successive polls.
+            poll_timeout (float): The maximum time that will wait before this operation is timed
+                out. By default, this will never time out.
+            cancel_on_termination (bool): Whether to cancel a sync in Airbyte if the Dagster runner is terminated.
+                This may be useful to disable if using Airbyte sources that cannot be cancelled and
+                resumed easily, or if your Dagster deployment may experience runner interruptions
+                that do not impact your Airbyte deployment.
+
+        Returns:
+            :py:class:`~AirbyteOutput`:
+                Details of the sync job.
+        """
+        connection_details = self.get_connection_details(connection_id)
+        start_job_details = self.start_sync_job(connection_id)
+        job = AirbyteJob.from_job_details(job_details=start_job_details)
+
+        self._log.info(f"Job {job.id} initialized for connection_id={connection_id}.")
+        poll_start = datetime.now()
+        poll_interval = (
+            poll_interval if poll_interval is not None else DEFAULT_POLL_INTERVAL_SECONDS
+        )
+        try:
+            while True:
+                if poll_timeout and datetime.now() > poll_start + timedelta(seconds=poll_timeout):
+                    raise Failure(
+                        f"Timeout: Airbyte job {job.id} is not ready after the timeout"
+                        f" {poll_timeout} seconds"
+                    )
+
+                time.sleep(poll_interval)
+                poll_job_details = self.get_job_details(job.id)
```

Comment:
nit; why make the caller initialize the job from the details. Feels like we should just return it directly